app/src/ui/open-pull-request/open-pull-request-header.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react'
+import { Branch } from '../../models/branch'
+import { DialogHeader } from '../dialog/header'
+import { createUniqueId } from '../lib/id-pool'
+import { Ref } from '../lib/ref'
+
+interface IOpenPullRequestDialogHeaderProps {
+  /** The base branch of the pull request */
+  readonly baseBranch: Branch
+
+  /** The branch of the pull request */
+  readonly currentBranch: Branch
+
+  /** The count of commits of the pull request */
+  readonly commitCount: number
+
+  /**
+   * Event triggered when the dialog is dismissed by the user in the
+   * ways described in the dismissable prop.
+   */
+  readonly onDismissed?: () => void
+}
+
+/**
+ * A header component for the open pull request dialog. Made to house the
+ * base branch dropdown and merge details common to all pull request views.
+ */
+export class OpenPullRequestDialogHeader extends React.Component<
+  IOpenPullRequestDialogHeaderProps,
+  {}
+> {
+  public render() {
+    const title = __DARWIN__ ? 'Open a Pull Request' : 'Open a pull request'
+    const { baseBranch, currentBranch, commitCount, onDismissed } = this.props
+    const commits = `${commitCount} commit${commitCount > 1 ? 's' : ''}`
+
+    return (
+      <DialogHeader
+        title={title}
+        titleId={createUniqueId(`Dialog_${title}_${title}`)}
+        dismissable={true}
+        onDismissed={onDismissed}
+      >
+        <div className="break"></div>
+        <div className="base-branch-details">
+          Merge {commits} into <Ref>{baseBranch.name}</Ref> from{' '}
+          <Ref>{currentBranch.name}</Ref>.
```

Comment:
I used the <Ref> component to highlight the branch names like it is on a pull request on dotcom. But, our merge branch dialog bolds branch names so I am happy to change it if anyone feels that bolding is more inline with Desktop and thus we should do that. (I am more inclined to think we should use the <Ref> component everywhere in desktop for branch names and commits when in a sentence).

![image](https://user-images.githubusercontent.com/75402236/189695944-a15592e8-9741-4deb-b3d2-d2dbf65f3621.png)

Reply:
Since in this case the branch names are not in the title but in the subtitle, I _think_ it makes sense to show them within `Ref` components. Or, at least, it doesn't feel that odd 😅